File: Week 01/id_477/1.两数之和.go
package leetcode

/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	tgtMap := make(map[int]int)
	for i, num := range nums {
		if idx, ok := tgtMap[target-num]; ok {
			return []int{idx, i}
		}
		tgtMap[num] = i
	}
	return nil
}

// @lc code=end

File: Week 01/id_282/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i,x in enumerate(nums) :
            temp=nums[:i]
            if target-x in temp:
                j = temp.index(target -x)
                return [i,j]
# @lc code=end

